Services/TMS/Table/ilTMSTableGUI.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+require_once("Services/Table/classes/class.ilTable2GUI.php");
+require_once("Services/TMS/Table/TMSTableParentGUI.php");
+
+/**
+ * This is our base implementation for the ilias table gui.
+ * It is minimized only to show data.
+ * Configuration always happens in parent gui
+ *
+ * @author Stefan Hecken	<stefan.hecken@concepts-and-training.de>
+ */
+final class ilTMSTableGUI extends ilTable2GUI
+{
+	/**
+	 * @var \Closure
+	 */
+	protected $fill_row;
+
+	/**
+	 * @var string
+	 */
+	protected $table_id;
+
+	public function __construct(TMSTableParentGUI $parent_gui, $parent_cmd, \Closure $fill_row, $table_id)
```

Comment:
\Closure fill_row ist hier doch redundant, weil man das schon aus der parent gui erhält.

Reply:
selbiges zur Table-id

Reply:
Und parent_cmd

Reply:
Das ist mit Absicht so. Die Table2GUI erwartet in ihrem Konstruktor eine parent gui. Deshalb muss ich die übergeben. Damit ich nun die an die anderen 3 Werte komme, müssten diese Funktionen in der Abstraction public sein. Das möchte ich aber nicht, denn außerhalb dieses Systems muss keine Zugriff auf diese Werte haben.

Wenn die Table2 nicht den Parent benötigen würde, dann würde dieser Parameter im Konstruktor fehlen.